src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.slice(-1)[0];
+  const toSeparator = toFormat.slice(-1)[0];
+  const dateElements = date.split(fromSeparator);
+
+  const dateMap = fromFormat.reduce((acc, format, index) => {
+    acc[format] = dateElements[index];
+
+    return acc;
+  }, {});
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    if (format === 'YY') {
+      return year.slice(-2);
+    }
+
+    return year;
+  };
+
+  const formattedDateParts = toFormat.map((format) => {
+    if (format === 'YY' || format === 'YYYY') {
+      return convertYear(dateMap['YYYY'] || dateMap['YY'], format);
+    }
+
+    return dateMap[format];
+  });
+
+  return formattedDateParts.filter(Boolean).join(toSeparator);
```

Comment:
Using `.filter(Boolean)` to remove falsy values is clever, but given the task's constraints, it's unnecessary since there should not be any undefined or empty string values in `formattedDateParts`. You can directly join the parts without filtering.